core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -4882,4 +4924,113 @@ private boolean isH2DB(Connection dbConnection) throws Exception {
+     * @throws SQLException if error occurred while retrieving database connection
+     */
+    protected Connection getDBConnection() throws UserStoreException, SQLException {
+
+        Connection dbConnection = null;
+
+        switch (jdbcConnectionCircuitBreakerState) {
+            case CIRCUIT_STATE_OPEN:
+                long circuitOpenDuration = System.currentTimeMillis() - thresholdStartTime;
+                if (circuitOpenDuration >= thresholdTimeoutInMilliseconds) {
+                    try {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Trying to obtain JDBC connection for " + getMyDomainName() +" domain" +
+                                    " when circuit breaker state: "
+                                    + jdbcConnectionCircuitBreakerState + " and circuit breaker open duration: "
+                                    + circuitOpenDuration + "ms.");
+                        }
+                        dbConnection = getConnection();
+                        jdbcConnectionCircuitBreakerState = CIRCUIT_STATE_CLOSE;
+                        thresholdStartTime = 0;
+                        break;
+                    } catch (Exception e) {
+                        log.error("Error occurred while obtaining connection", e);
+                        thresholdStartTime = System.currentTimeMillis();
+                        if (log.isDebugEnabled()) {
+                            log.debug("JDBC connection circuit breaker state set to: "
+                                    + jdbcConnectionCircuitBreakerState);
+                        }
+                        throw new CircuitBreakerException("Error occurred while obtaining connection.", e);
+                    }
+                } else {
+                    throw new CircuitBreakerException(
+                            "JDBC Connection circuit breaker is in open state for " + circuitOpenDuration
+                                    + "ms and has not reach the threshold timeout: " + thresholdTimeoutInMilliseconds
+                                    + "ms, hence avoid establishing the connection for " + getMyDomainName() +
+                                    " domain");
+                }
+            case CIRCUIT_STATE_CLOSE:
+                try {
+                    if (log.isDebugEnabled()) {
+                        log.debug("connection circuit breaker state: " + jdbcConnectionCircuitBreakerState
+                                + ", so trying to obtain the connection for " + getMyDomainName() + " domain");
+                    }
+                    dbConnection = getConnection();
+                    break;
+                } catch (Exception e) {
+                    log.error("Error occurred while obtaining JDBC connection for " + getMyDomainName() +
+                            " domain", e);
+                    log.error("Trying again to get connection.");
```

Comment:
is this a error log?

Reply:
The second error log can be moved to a debug log.